pkg/cli/root_test.go

```diff
@@ -1018,11 +1018,56 @@ func TestVerify(t *testing.T) {
 	})
 }
 
+func TestAutoPortSelection(t *testing.T) {
+	Convey("Run server without specifying a port (port 0)", t, func() {
+		logFile, err := os.CreateTemp("", "zot-log*.txt")
+		logPath := logFile.Name()
+		So(err, ShouldBeNil)
+
+		content := fmt.Sprintf(`{"distSpecVersion": "1.0.0", "storage": {"rootDirectory": "/tmp/zot"},
+		"http": {"address": "127.0.0.1", "port": "0"},
+		"log": {"level": "debug", "output": "%s"}}`, logFile.Name())
+
+		cfgfile, err := os.CreateTemp("", "zot-test*.json")
+		So(err, ShouldBeNil)
+
+		defer os.Remove(cfgfile.Name()) // clean up
+
+		_, err = cfgfile.Write([]byte(content))
+		So(err, ShouldBeNil)
+
+		os.Args = []string{"cli_test", "serve", cfgfile.Name()}
+		go func() {
+			err = cli.NewServerRootCmd().Execute()
+			So(err, ShouldBeNil)
+		}()
+
+		_, err = ReadLogFileAndSearchString(logPath, "CVE config not provided, skipping CVE update", 90*time.Second)
```

Comment:
Why wait for the CVE related string?
Wait for the port related log message (`HTTP port `). That is what you are testing after all.

Reply:
I made the changes